### PR TITLE
Moved convolutions to C# modules

### PR DESCRIPTION
**Pull request by shaltielshmid**

### CODE REVIEW

**Comment by shaltielshmid** [comment]
One point I'd like to note: In PyTorch they have one call "F.conv1d", where they allow either a string or an int/tuple. To replicate this behavior, I did two things:
1] In the module, there is a specific function for "SetPadding" instead of making the properties modifiable, and there are two properties - one "padding" which is a long[] and one "padding_type" which is the enum. 
2] In the torch.nn.functional class, I split it into two functions - conv1d and conv1d_padding, following the naming notation in LibTorch (see [here](https://github1s.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Convolution.cpp#L1069-L1070)). 

Aside from that - looking forward to hearing your thoughts!
